cognite/neat/core/extractors/rules_to_graph_capturing_sheet.py

```diff
@@ -0,0 +1,153 @@
+import logging
+import uuid
+import warnings
+from pathlib import Path
+
+import xlsxwriter
+
+from cognite.neat.core.data_classes.transformation_rules import TransformationRules
+from cognite.neat.core.extractors.rules_to_graphql import repair_name as to_graphql_name
+
+
+def _get_column(column_number: int) -> str:
+    """Converts a column number to a letter"""
+    dividend = column_number
+    column_letter = ""
+    while dividend > 0:
+        modulo = (dividend - 1) % 26
+        column_letter = chr(65 + modulo) + column_letter
+        dividend = int((dividend - modulo) / 26)
+    return column_letter
+
+
+def _get_header_style(**kwargs):
+    """Return the header style"""
+    header_cfg = {
+        "border": kwargs.get("border", 1),
+        "bg_color": kwargs.get("bg_color", "#2FB5F2"),
+        "bold": kwargs.get("bold", True),
+        "text_wrap": kwargs.get("text_wrap", False),
+        "valign": kwargs.get("valign", "vcenter"),
+        "indent": kwargs.get("indent", 1),
+        "font_size": kwargs.get("font_size", 16),
+        "font_name": kwargs.get("font_name", "Helvetica"),
+    }
+
+    identifier_header_cfg = header_cfg.copy()
+    identifier_header_cfg["bg_color"] = "#FFB202"
+
+    return header_cfg, identifier_header_cfg
+
+
+def _add_uuid_identifiers(sheets, sheet_name, no_rows):
+    """Adds UUID-based auto identifier to a sheet"""
+    for i in range(no_rows):
+        fixed_name = to_graphql_name(sheet_name, "class", True)
+        sheets[sheet_name].write_formula(f"A{i+2}", f'=IF(ISBLANK(B{i+2}), "","{fixed_name}-{uuid.uuid4()}")')
+
+
+def _add_index_identifiers(sheets, sheet_name, no_rows):
+    """Adds index-based auto identifier to a sheet"""
+    for i in range(no_rows):
+        fixed_name = to_graphql_name(sheet_name, "class", True)
+        sheets[sheet_name].write_formula(f"A{i+2}", f'=IF(ISBLANK(B{i+2}), "","{fixed_name}-{i+1}")')
+
+
+def _set_column_width(sheet, no_columns, width):
+    """Sets the width of all columns in a sheet"""
+    for i in range(no_columns + 1):
+        sheet.set_column(i, i, width)
+
+
+def _add_drop_down_list(sheets, sheet_name, column, no_rows, value_sheet, value_column):
+    """Adds a drop down list to a column"""
+    logging.info(f"Adding drop down list to <{sheet_name}!{column}> with values from <{value_sheet}!{value_column}>")
+    for i in range(no_rows):
+        sheets[sheet_name].data_validation(
+            f"{column}{i+2}",
+            {
+                "validate": "list",
+                "source": f"={value_sheet}!{value_column}2:{value_column}{no_rows}",
+            },
+        )
+
+
+def rules2graph_capturing_sheet(
+    file_path: Path,
+    transformation_rules: TransformationRules,
```

Comment:
I think these should be in the reverse order. As I understand it input=`transformation_rules` and output=`file_path`, input before output. Furthermore, I would rename `file_path` to `output_capturing_sheet_path`.

Reply:
Good point! Done !